### ISSUE TITLE
Bug: <Persona 4 Golden (MS Store) not detecting mods> because <the .exe name is different than the Steam's one>

### ISSUE THREAD

**Issue by Randy28CR** [Reloaded-Project/Reloaded-II#338]
**Describe the Bug**
The Persona 4 Golden (Xbox/MS Store) is not detecting mods because it has a different .exe name (p4pc_DT_mc.exe) than Steam's .exe (p4g.exe). Due to this, Reloaded II does not detect the "p4pc_DT_mc.exe" as P4G, so mods are not aplicable. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'Reloaded II'
2. Click on 'Add an application'
3. Search 'p4pc_DT_mc.exe' in the MS Store games folder
4. Try to download or select P4G mods to it. They will not show up.

**Expected Behaviour**
Reload II shows up P4G mods after adding the Persona 4 Golden MS Store .exe

**Additional Context**
Reload II injects and launches P4G MS Store as expected, so I think it has to be someway to add "p4pc_DT_mc.exe" as name attached to Persona 4 Golden besides "p4g.exe", this way Reloaded II can actually link P4G mods and thus they work in that version.

Persona 5 Royal and Persona 3 Reload have the same .exe name on both MS store and Steam versions, so after lastest Reloaded II patch, those work completely as expected. Only P4G has issues and it is due to having a different .exe name than Steam's version.

**Comment by Sewer56** [comment]
A temporary workaround for this is to open `AppConfig.json` for `p4pc_DT_mc.exe` and change AppId to `p4g.exe`.

Not ideal, I know. Normally something like this would be handled via the [Community Repo](https://github.com/Reloaded-Project/Reloaded.Community), though the feature to alias an `AppId` was never added there. 

I don't think I'd want to do it myself, as R2 is in maintenance mode, given that I'm working on a successor. I could maybe hardcode an override into the launcher as a temporary hack though; since it's unlikely a game that's not Persona would have `p4pc_DT_mc` as its binary name.

**Comment by Randy28CR** [comment]
> A temporary workaround for this is to open `AppConfig.json` for `p4pc_DT_mc.exe` and change AppId to `p4g.exe`.
> 
> Not ideal, I know. Normally something like this would be handled via the [Community Repo](https://github.com/Reloaded-Project/Reloaded.Community), though the feature to alias an `AppId` was never added there.
> 
> I don't think I'd want to do it myself, as R2 is in maintenance mode, given that I'm working on a successor. I could maybe hardcode an override into the launcher as a temporary hack though; since it's unlikely a game that's not Persona would have `p4pc_DT_mc` as its binary name.

Already did and, even though it now shows up the mods, I got this error message. I tried to launch it both with and without mods enabled. Same results.

Tbh, I am not that good at programing and stuff like that. I can understand basic things, so idk what those error mean.


![imagen_2024-03-13_113549722](https://github.com/Reloaded-Project/Reloaded-II/assets/163334690/fdfd950e-aeb9-480a-8ff4-2e14868ac6df)

I would love some help as I think this may be something fixable, but i understand if you can´t or want. I bought MS Store versions instead of Steam cause I also have Xbox and they have Xbox Play Anywhere,

**Comment by Sewer56** [comment]
Ah, I guess I should probably restore the code for symlink resolution on GamePass titles. I suppose that without AppId matching EXE name, location would still be the go-to.

In the meantime, you can manually copy the path from `Current Process Location` in log to your `AppConfig.json`'s EXE path, should fix your issue.

I'll throw it in a hotfix in the weekend.

**Comment by Randy28CR** [comment]
> Ah, I guess I should probably restore the code for symlink resolution on GamePass titles. I suppose that without AppId matching EXE name, location would still be the go-to.
> 
> In the meantime, you can manually copy the path from `Current Process Location` in log to your `AppConfig.json`'s EXE path, should fix your issue.
> 
> I'll throw it in a hotfix in the weekend.

Now it works with that suggestion you gave me!
But there is just a but: I copied the Current Process Location in log to the AppConfig.json and it didn´t work.
"C:\Program Files\WindowsApps\SEGAofAmericaInc.D0cb6b3aet_1.0.34.0_x64__s751p9cej88mt\p4pc_DT_mc.exe"

Then I noticed that, into AppConfig,json, the paths had a double "\\", so I added one per location:
"C:\\Program Files\\WindowsApps\\SEGAofAmericaInc.D0cb6b3aet_1.0.34.0_x64__s751p9cej88mt\\p4pc_DT_mc.exe"

After that, Persona 4 Golden MS Store version FINALLY works on Reloaded II as intended.

Thank you so much, you are awesome!
I will be waiting for that hotfix anyway and I will donate something to the project as soon as I can.

Regards!
